src/gpu/gpu-device.ts

```diff
@@ -0,0 +1,140 @@
+import { Worker } from 'node:worker_threads';
+
+import { JSDOM } from 'jsdom';
+import {
+    // components
+    AnimComponentSystem,
+    RenderComponentSystem,
+    CameraComponentSystem,
+    LightComponentSystem,
+    GSplatComponentSystem,
+    ScriptComponentSystem,
+    // handlers
+    AnimClipHandler,
+    AnimStateGraphHandler,
+    BinaryHandler,
+    ContainerHandler,
+    CubemapHandler,
+    GSplatHandler,
+    RenderHandler,
+    TextureHandler,
+    // rest
+    PIXELFORMAT_BGRA8,
+    AppBase,
+    AppOptions,
+    Texture,
+    WebgpuGraphicsDevice
+} from 'playcanvas/debug';
+import { create, globals } from 'webgpu';
+
+Object.assign(globalThis, globals);
+
+const jsdomSetup = () => {
+    const html = '<!DOCTYPE html><html><head></head><body></body></html>';
+
+    const jsdom = new JSDOM(html, {
+        resources: 'usable',         // Allow the engine to load assets
+        runScripts: 'dangerously',   // Allow the engine to run scripts
+        url: 'http://localhost:3000' // Set the URL of the document
+    });
+
+    // Copy the window and document to global scope
+    // @ts-ignore
+    global.window = jsdom.window;
+    global.document = jsdom.window.document;
+
+    // Copy the DOM APIs used by the engine to global scope
+    global.ArrayBuffer = jsdom.window.ArrayBuffer;
+    global.Audio = jsdom.window.Audio;
+    global.DataView = jsdom.window.DataView;
+    global.Image = jsdom.window.Image;
+    global.KeyboardEvent = jsdom.window.KeyboardEvent;
+    global.MouseEvent = jsdom.window.MouseEvent;
+    global.XMLHttpRequest = jsdom.window.XMLHttpRequest;
+};
+
+class Application extends AppBase {
+    constructor(canvas: HTMLCanvasElement, options: any = {}) {
+        super(canvas);
+
+        const appOptions = new AppOptions();
+
+        appOptions.graphicsDevice = options.graphicsDevice;
+
+        appOptions.componentSystems = [
+            AnimComponentSystem,
+            CameraComponentSystem,
+            GSplatComponentSystem,
+            LightComponentSystem,
+            RenderComponentSystem,
+            ScriptComponentSystem
+        ];
+
+        appOptions.resourceHandlers = [
+            AnimClipHandler,
+            AnimStateGraphHandler,
+            BinaryHandler,
+            ContainerHandler,
+            CubemapHandler,
```

Comment:
I suspect you could also remove some componentSystems and resourceHandles, but that has low impact really.